File: sdk/java/src/main/java/com/xxxxteam/tempmail/providers/MailcatAi.java
package com.xxxxteam.tempmail.providers;

import com.google.gson.JsonElement;
import com.google.gson.JsonObject;
import com.xxxxteam.tempmail.Email;
import com.xxxxteam.tempmail.EmailInfo;
import com.xxxxteam.tempmail.HttpResult;
import com.xxxxteam.tempmail.HttpClient;
import com.xxxxteam.tempmail.Json;
import com.xxxxteam.tempmail.Normalizer;

import java.util.ArrayList;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

/**
 * mailcat.ai 渠道 — https://mailcat.ai
 *
 * <p>POST /mailboxes 创建邮箱（无 body），返回 Bearer token。
 * GET /inbox 获取邮件列表。</p>
 */
public final class MailcatAi {

    private static final String BASE_URL = "https://api.mailcat.ai";

    private MailcatAi() {
    }

    private static Map<String, String> headers() {
        Map<String, String> h = new LinkedHashMap<>();
        h.put("User-Agent", "Mozilla/5.0");
        h.put("Accept", "application/json");
        return h;
    }

    /**
     * 创建 mailcat.ai 临时邮箱。
     *
     * @return 邮箱信息
     */
    public static EmailInfo generate() {
        HttpResult resp = HttpClient.post(BASE_URL + "/mailboxes", null, null, headers());
        resp.ensureSuccess();

        JsonObject body = Json.parseObject(resp.getBody());
        if (body == null) throw new RuntimeException("mailcat-ai: 响应格式无效");

        JsonObject data = body.has("data") && body.get("data").isJsonObject()
                ? body.getAsJsonObject("data") : null;
        if (data == null) throw new RuntimeException("mailcat-ai: 响应 data 格式无效");

        String address = Json.str(data, "email");
        String token = Json.str(data, "token");
        if (address.isEmpty()) throw new RuntimeException("mailcat-ai: 缺少邮箱地址");
        if (token.isEmpty()) throw new RuntimeException("mailcat-ai: 缺少认证令牌");

        return new EmailInfo("mailcat-ai", address, token, null, null);
    }

    /**
     * 获取邮件列表。
     *
     * @param token Bearer token
     * @param email 邮箱地址
     * @return 邮件列表
     */
    public static List<Email> getEmails(String token, String email) {
        Map<String, String> h = headers();
        h.put("Authorization", "Bearer " + (token != null ? token : ""));

        HttpResult resp = HttpClient.get(BASE_URL + "/inbox", h);
        if (resp.getStatusCode() == 404) return new ArrayList<>();
        resp.ensureSuccess();

        JsonObject body = Json.parseObject(resp.getBody());
        if (body == null) return new ArrayList<>();

        var messages = Json.arr(body, "data");
        if (messages == null) return new ArrayList<>();

        List<Email> out = new ArrayList<>();
        for (JsonElement item : messages) {
            if (item.isJsonObject()) {
                out.add(Normalizer.normalizeEmail(Json.toDict(item), email));
            }
        }
        return out;
    }
}

File: sdk/kotlin/src/main/kotlin/com/xxxxteam/tempmail/providers/MailcatAi.kt
package com.xxxxteam.tempmail.providers

import com.xxxxteam.tempmail.Email
import com.xxxxteam.tempmail.EmailInfo
import com.xxxxteam.tempmail.Normalize
import com.xxxxteam.tempmail.Provider
import kotlinx.serialization.json.*

/**
 * mailcat.ai 渠道（mailcat.ai）。
 *
 * POST /mailboxes 创建邮箱（无 body），返回 Bearer token；
 * GET /inbox 获取邮件列表。
 */
object MailcatAi : Provider {

    private const val CHANNEL = "mailcat-ai"
    private const val BASE_URL = "https://api.mailcat.ai"
    private val headers = mapOf(
        "User-Agent" to "Mozilla/5.0",
        "Accept" to "application/json",
    )

    /** 创建临时邮箱。 */
    override suspend fun generate(): EmailInfo {
        val resp = ProviderUtil.httpPost("$BASE_URL/mailboxes", null, null, headers)
        resp.ensureSuccess()
        val body = ProviderUtil.parseObject(resp.body) ?: throw RuntimeException("mailcat-ai: 响应格式无效")
        val data = body["data"] as? JsonObject ?: throw RuntimeException("mailcat-ai: 响应 data 格式无效")
        val address = ProviderUtil.str(data, "email")
        val token = ProviderUtil.str(data, "token")
        if (address.isEmpty()) throw RuntimeException("mailcat-ai: 缺少邮箱地址")
        if (token.isEmpty()) throw RuntimeException("mailcat-ai: 缺少认证令牌")
        return EmailInfo(email = address, channel = CHANNEL, token = token)
    }

    /** 获取邮件列表。 */
    override suspend fun getEmails(info: EmailInfo): List<Email> {
        val h = headers + ("Authorization" to "Bearer ${info.token}")
        val resp = ProviderUtil.httpGet("$BASE_URL/inbox", h)
        if (resp.statusCode == 404) return emptyList()
        resp.ensureSuccess()
        val body = ProviderUtil.parseObject(resp.body) ?: return emptyList()
        val messages = ProviderUtil.arr(body, "data") ?: return emptyList()
        return messages.filterIsInstance<JsonObject>().map { Normalize.fromJson(it, info.email) }
    }
}
